### PR TITLE
Log xDS node ID with verbosity INFO

### PR DESCRIPTION
**Pull request by mohanli-ml**
Log xDS node ID so that we can better debug DirectPath issues.

C++ change: https://github.com/grpc/grpc/pull/31797.

### CODE REVIEW

**Review by ejona86 — COMMENTED** [review]
This needs to happen in XdsClient or Bootstrap; this code won't be run once federation is available. Based on http://b/260634412#comment15 I had expected it to be done in XdsClient (XdsClientImpl).

File: xds/src/main/java/io/grpc/xds/XdsClientImpl.java [hunk 1]
```diff
@@ -60,13 +60,18 @@
...
+import java.util.logging.Level;
+import java.util.logging.Logger;
 import javax.annotation.Nullable;
 
 /**
  * XdsClient implementation for client side usages.
  */
 final class XdsClientImpl extends XdsClient implements XdsResponseHandler, ResourceStore {
 
+  static boolean logXdsNodeId = Boolean.parseBoolean(System.getenv("GRPC_LOG_XDS_NODE_ID"));
```
Comment (ejona86) [inline xds/src/main/java/io/grpc/xds/XdsClientImpl.java:72]: Make both of these private.

This should be named with the LOG_XDS_NODE_ID style as it is immutable. (Logger is not immutable, so it uses the normal naming.)

**Review by ejona86 — COMMENTED** [review]
Minor stuff, then looks good.

File: xds/src/main/java/io/grpc/xds/XdsClientImpl.java (hunk 1 above)
Comment (mohanli-ml) [reply xds/src/main/java/io/grpc/xds/XdsClientImpl.java]: Done. PTAL!

**Review by ejona86 — APPROVED** [review]

**Pull request merged**